contracts/zero-ex/tests/Multiplex.t.sol

```diff
@@ -0,0 +1,1453 @@
+    }
+
+    function makeUniswapV3BatchSubcall(
+        address[] memory tokens,
+        uint256 sellAmount
+    ) private pure returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.UniswapV3,
+                sellAmount: sellAmount,
+                data: encodePathUniswapV3(tokens)
+            });
+    }
+
+    function makeLiquidityProviderMultiHopSubcall()
+        private
+        view
+        returns (IMultiplexFeature.MultiHopSellSubcall memory)
+    {
+        return
+            IMultiplexFeature.MultiHopSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.LiquidityProvider,
+                data: abi.encode(address(liquidityProvider), hex"")
+            });
+    }
+
+    function makeLiquidityProviderBatchSubcall(
+        uint256 sellAmount
+    ) private view returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.LiquidityProvider,
+                sellAmount: sellAmount,
+                data: abi.encode(address(liquidityProvider), hex"")
+            });
+    }
+
+    function makeTransformERC20Subcall(
+        IERC20TokenV06 inputToken,
+        IERC20TokenV06 outputToken,
+        uint256 sellAmount,
+        uint256 mintAmount
+    ) private view returns (IMultiplexFeature.BatchSellSubcall memory) {
+        ITransformERC20Feature.Transformation[] memory transformations = new ITransformERC20Feature.Transformation[](1);
+        transformations[0] = ITransformERC20Feature.Transformation(
+            uint32(transformerNonce),
+            abi.encode(address(inputToken), address(outputToken), 0, mintAmount, 0)
+        );
+
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.TransformERC20,
+                sellAmount: sellAmount,
+                data: abi.encode(transformations)
+            });
+    }
+
+    function makeNestedBatchSellSubcall(
+        IMultiplexFeature.BatchSellSubcall[] memory calls
+    ) private pure returns (IMultiplexFeature.MultiHopSellSubcall memory) {
+        return
+            IMultiplexFeature.MultiHopSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.BatchSell,
+                data: abi.encode(calls)
+            });
+    }
+
+    function makeNestedMultiHopSellSubcall(
+        address[] memory tokens,
+        IMultiplexFeature.MultiHopSellSubcall[] memory calls,
+        uint256 sellAmount
+    ) private pure returns (IMultiplexFeature.BatchSellSubcall memory) {
+        return
+            IMultiplexFeature.BatchSellSubcall({
+                id: IMultiplexFeature.MultiplexSubcall.MultiHopSell,
+                sellAmount: sellAmount,
+                data: abi.encode(tokens, calls)
+            });
+    }
+
+    function makeArray(
```

Comment:
Do we really need this many specialized versions of this pattern? Does this get reused sufficiently often that it warrants being separated out? This is a lot of code to scroll through.

Reply:
The idea here was to increase readability in each of the test cases. We don't need it, but imo it helps readability to be able to initialize dynamic arrays inline. It lets us express the tests more concisely and clearly than we otherwise could.

Some of these `makeArray` functions specific to objects used only in `MultiplexFeature` wouldn't make sense to move outside of this file, but some of the more generic `makeArray` functions would ideally be in some helper code shared by all tests.